nautobot/extras/api/serializers.py

```diff
@@ -142,6 +144,16 @@ class NautobotModelSerializer(
     """
 
 
+class RoleSerializerField(LimitQuerysetChoicesSerializerMixin, NestedRoleSerializer):
+    """NestedSerializer field for `Role` object fields."""
+
+
+class RoleModelSerializerMixin(BaseModelSerializer):
+    """Mixin to add `role` choice field to model serializers."""
+
+    role = RoleSerializerField(required=False)
```

Comment:
Do we need separate mixins for models for which role is required versus models that permit a null role?